src/tree.h

```diff
@@ -42,6 +42,8 @@ extern int git_tree_entry_icmp(const git_tree_entry *e1, const git_tree_entry *e
 void git_tree__free(void *tree);
 int git_tree__parse(void *tree, git_odb_object *obj);
 
+int git_treebuilder_write_buffered(git_oid *oid, git_treebuilder *bld, git_buf *shared_buf);
```

Comment:
The name `_buffered` sounds like it's related to buffered I/O, or that the buffer is the output. Something like `_with_buffer` would be more in line with our naming.